e/php/bootstrap

```diff
@@ -0,0 +1,171 @@
+#!/opt/bin/php -c/opt/php.ini
+<?php
+use Bref\Bridge\Psr7\RequestFactory;
+use Bref\Http\LambdaResponse;
+use Psr\Http\Message\RequestInterface;
+
+use hollodotme\FastCGI\Client;
+use hollodotme\FastCGI\SocketConnections\NetworkSocket;
+use Symfony\Component\Process\Process;
+
+echo 'Cold start' . PHP_EOL;
+
+ini_set('display_errors', '1');
+error_reporting(E_ALL);
+
+$lambdaRuntimeApi = getenv('AWS_LAMBDA_RUNTIME_API');
+
+require __DIR__ . '/vendor/autoload.php';
+
+class PsrRequest extends \hollodotme\FastCGI\Requests\AbstractRequest {
+    /** @var  RequestInterface */
+    protected $request;
+    public function __construct(string $scriptFilename, RequestInterface $request)
+    {
+        $this->request = $request;
+        parent::__construct($scriptFilename, (string) $request->getBody());
+
+        $this->setRequestUri((string) $request->getUri());
+    }
+
+    public function getRequestMethod(): string
+    {
+        return $this->request->getMethod();
+    }
+}
+
+
+$cgi = new Process("php-cgi -b 127.0.0.1:9000 -c /opt/php.ini -d extension_dir=/opt/php/extensions");
```

Comment:
Wow so I had no idea this existed. If I understand correctly this is the `php-cgi` binary (which is usually used as CGI protocol, i.e. `exec`) but with the `-b` flag it turns into a FastCGI server?

For scenario E I thought we would try PHP-FPM, I didn't know about this execution mode. Do you know how it compares to PHP-FPM?

What happens to the `php-cgi` process: is it spawning children, or is it processing one request at a time via the FastCGI interface?

Reply:
**CGI SAPI** basically spawns a single process for each request and just takes the input from stdin. These processes are typically owned by the executing service.

**FastCGI SAPI** gets communication over a socket. Instead of creating a new process for each request, FastCGI uses persistent processes to handle a series of requests. These processes are typically owned by a FastCGI server, not the executing service. The management of these processes is somewhat rudimentary, but it scales a lot better than vanilla CGI.

**FPM** is a process manager to manage the **FastCGI SAPI**, basically, this overcomes the rudimentary process management that you get from simply attempting to use **FastCGI SAPI** alone.

For our needs, **FPM** is likely heavier than we need, due to the fact that we only ever use a single process at a time. **FastCGI SAPI** will easily provide a single, reusable process. **CGI SAPI** also, necessarily, provides a single, non-reusable process.

Reply:
I did not see a performance difference on mac between the built-in server, fastcgi and fpm, but fpm requires more configuration so harder to setup.

Reply:
Awesome, thanks for the explanations. This implementation sounds like the best then!

Reply:
Yeah except that I can't get it to work with Symfony :P Probably just because the mapping of the headers, but feels like the built-in server is better suited for HTTP requests.

Reply:
If we can get this merged, I could more easily contribute to it. I think the current PHP library you are using will work just fine, just needs a bit more configuration if I read it correctly.

Reply:
I have it working in my fork, you can see the [bootstrap](https://github.com/stechstudio/bref-bootstrap-benchmarks/blob/develop/e.1/symfony/bootstrap). Both the PHP and Symfony tests are running locally for me.

Reply:
Yeah locally everything is working fine for me also, just nog on AWS itself.

Reply:
Just tested mine on AWS https://pgnuhy0hpa.execute-api.us-east-2.amazonaws.com/Prod and it appears to be working fine.

Reply:
Okay will give it a go and compare differences. But if yours is working it's fine for the benchmark :)

Reply:
I can merge this if necessary, I'm just holding off because it's marked as `WIP`